AutomatedTesting/Gem/PythonTests/atom_renderer/atom_hydra_scripts/hydra_AtomEditorComponents_DirectionalLightAdded.py

```diff
@@ -0,0 +1,148 @@
+    Creation and deletion undo/redo should also work.
+
+    Test Steps:
+    1) Create a Directional Light entity with no components.
+    2) Add Directional Light component to Directional Light entity.
+    3) UNDO the entity creation and component addition.
+    4) REDO the entity creation and component addition.
+    5) Enter/Exit game mode.
+    6) Test IsHidden.
+    7) Test IsVisible.
+    8) Add Camera entity.
+    9) Add Camera component to Camera entity
+    10) Set the Directional Light component property Shadow|Camera to the Camera entity.
+    11) Delete Directional Light entity.
+    12) UNDO deletion.
+    13) REDO deletion.
+    14) Look for errors.
+
+    :return: None
+    """
+
+    import azlmbr.bus as bus
+    import azlmbr.editor as editor
+    import azlmbr.legacy.general as general
+    import azlmbr.math as math
+
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper as helper
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        helper.init_idle()
+        helper.open_level("Physics", "Base")
+
+        # Test steps begin.
+        # 1. Create a Directional Light entity with no components.
+        directional_light_name = "Directional Light"
+        directional_light_entity = EditorEntity.create_editor_entity_at(
+            math.Vector3(512.0, 512.0, 34.0), f"{directional_light_name}")
+        Report.critical_result(Tests.directional_light_creation, directional_light_entity.exists())
+
+        # 2. Add Directional Light component to Directional Light entity.
+        directional_light_component = directional_light_entity.add_component(directional_light_name)
+        Report.critical_result(
+            Tests.directional_light_component, directional_light_entity.has_component(directional_light_name))
+
+        # 3. UNDO the entity creation and component addition.
+        # Requires 3 UNDO to remove the Entity completely.
+        for x in range(4):
+            general.undo()
+        Report.result(Tests.creation_undo, not directional_light_entity.exists())
+
+        # 4. REDO the entity creation and component addition.
+        # Requires 3 REDO calls to remove the Entity completely.
+        for x in range(4):
+            general.redo()
+        Report.result(Tests.creation_redo, directional_light_entity.exists())
+
+        # 5. Enter/Exit game mode.
+        helper.enter_game_mode(Tests.enter_game_mode)
+        general.idle_wait_frames(1)
+        helper.exit_game_mode(Tests.exit_game_mode)
+
+        # 6. Test IsHidden.
+        directional_light_entity.set_visibility_state(False)
+        is_hidden = editor.EditorEntityInfoRequestBus(bus.Event, 'IsHidden', directional_light_entity.id)
```

Comment:
repeated: is_hidden(), check for all the instances in your tests so can be updated